core/src/main/java/org/elasticsearch/index/query/MatchQueryBuilder.java

```diff
@@ -31,80 +41,90 @@
  */
 public class MatchQueryBuilder extends AbstractQueryBuilder<MatchQueryBuilder> {
 
+    /** The default name for the match query */
     public static final String NAME = "match";
 
-    public enum Type {
-        /**
-         * The text is analyzed and terms are added to a boolean query.
-         */
-        BOOLEAN,
-        /**
-         * The text is analyzed and used as a phrase query.
-         */
-        PHRASE,
-        /**
-         * The text is analyzed and used in a phrase query, with the last term acting as a prefix.
-         */
-        PHRASE_PREFIX
-    }
+    /** The default mode terms are combined in a match query */
+    public static final Operator DEFAULT_OPERATOR = Operator.OR;
 
-    public enum ZeroTermsQuery {
-        NONE,
-        ALL
-    }
+    /** The default mode match query type */
+    public static final MatchQuery.Type DEFAULT_TYPE = MatchQuery.Type.BOOLEAN;
 
-    private final String name;
+    private final String fieldName;
 
-    private final Object text;
+    private final Object value;
 
-    private Type type;
+    private MatchQuery.Type type = DEFAULT_TYPE;
 
-    private Operator operator;
+    private Operator operator = DEFAULT_OPERATOR;
 
     private String analyzer;
 
-    private Integer slop;
+    private int slop = MatchQuery.DEFAULT_PHRASE_SLOP;
 
-    private Fuzziness fuzziness;
+    private Fuzziness fuzziness = null;
```

Comment:
weird, I thought this was very similar to query_string where we do have AUTO as default, but eventually it isn't, sorry.